### ISSUE TITLE
Allow `react-native-vision-camera` `2.15.4` and `react-native-reanimated` `3.1.0` to work together

### ISSUE THREAD

**Issue by AndrewGable** [Expensify/App#22749]
## Problem
Install `react-native-vision-camera` `2.15.4` and `react-native-reanimated` `3.1.0` on the same React Native app and you will see this error when trying to run iOS via `npm run ios` 💥 

```
❌  /Users/andrew/Expensidev/App/node_modules/react-native-vision-camera/ios/Frame Processor/FrameProcessorRuntimeManager.mm:146:32: no member named 'ShareableValue' in namespace 'reanimated'

    auto worklet = reanimated::ShareableValue::adapt(runtime, arguments[1], runtimeManager.get());
```

**Note:** A patch was prepared by @mrousavy that fixes this, but it would be great to support this without a patch.

```diff 
diff --git a/node_modules/react-native-vision-camera/ios/Frame Processor/FrameProcessorRuntimeManager.mm b/node_modules/react-native-vision-camera/ios/Frame Processor/FrameProcessorRuntimeManager.mm
index 3841b20..687ea94 100644
--- a/node_modules/react-native-vision-camera/ios/Frame Processor/FrameProcessorRuntimeManager.mm
+++ b/node_modules/react-native-vision-camera/ios/Frame Processor/FrameProcessorRuntimeManager.mm
@@ -19,6 +19,8 @@
 #import <React/RCTUIManager.h>
 #import <ReactCommon/RCTTurboModuleManager.h>

+#define VISION_CAMERA_DISABLE_FRAME_PROCESSORS 1
+
 #ifndef VISION_CAMERA_DISABLE_FRAME_PROCESSORS
   #if __has_include(<RNReanimated/NativeReanimatedModule.h>)
     #if __has_include(<RNReanimated/RuntimeManager.h>)
```

## Solution
Allow the two libraries to work together without a patch.

**Comment by AndrewGable** [comment]
@mrousavy and @tomekzaw - Can you comment here so I can assign you? Thanks!

**Comment by mrousavy** [comment]
👋

**Comment by AndrewGable** [comment]
Updates being left in Slack!

**Comment by AndrewGable** [comment]
We just merged the patch to `main`, any update here from library maintainers side?

**Comment by tomekzaw** [comment]
@AndrewGable I've managed to get it working with RNVisionCamera v3. I need some more time to backport the changes to RNVisionCamera v2 and submit PRs to both libraries. Will keep you updated!

**Comment by mrousavy** [comment]
Awesome work @tomekzaw! Feel free to DM me on Slack anytime if you have any questions or want me to do any changes :)

**Comment by tomekzaw** [comment]
Hey! I've managed to integrate RNVisionCamera v2 with Reanimated v3 on both platforms (Android and iOS) 🥳 

Here's the PR to react-native-vision-camera (to the `v2` branch):

* https://github.com/mrousavy/react-native-vision-camera/pull/1759

I've used some commits from @mrousavy's PR which mostly bumps react-native version in the example app and does some general cleanup which served as a base for my further work on the integration. Also, Marc provided me with instructions on how to test out frame processors in the example app on Android. Thanks!

iOS demo: https://twitter.com/tomekzaw_/status/1697302708879765983
Android demo: https://twitter.com/tomekzaw_/status/1697604643365278077

I believe we could integrate the new RNVisionCamera v3 with Reanimated v3 this way. In fact, this should be fairly easy to do since we could re-use the migrated code from v2.

**Comment by mrousavy** [comment]
Let's go 🚀 

We're finishing the last bits on that PR and then I'll create a release on the v2 branch 👍

**Comment by AndrewGable** [comment]
@mrousavy - Do you think we could include this fix and https://github.com/mrousavy/react-native-vision-camera/pull/1782 on a V2 release? That'd be great for us 🙏

**Comment by mrousavy** [comment]
Yo Andrew - sure no problem I'll backport that!

**Comment by mrousavy** [comment]
Pushed that specific change and also a fix to build without frame processors using `disableFrameProcessors`. Released to npm @ `2.16.1`! 🚀 

Thanks again for your hard work @tomekzaw, this was just amazing help bro 🙏
